### DISCUSSION TITLE
[4.0 alpha] Move the "Showing your unsaved changes" and discard button to the top in slideouts

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#10611] [Ideas]
Curious what others think about this one, but in my opinion it's a bit odd that the new "Showing your unsaved changes" message (and the "Discard changes" button) sits at the bottom of the slideout panels, instead of at the top, like it does on edit pages.

Specifically, my worry is that 

a) The message is kinda hidden at the bottom of the panel, and probably more likely to go unnoticed by editors (the light blue color against the gray background doesn't exactly help it stand out, either)

b) Since the position isn't consistent between standalone edit pages and slideouts, the "unsaved changes" in slideouts might be interpreted by some authors as specifically "scoped" to the slideout panel.

I'll admit these aren't huge issues by any means – but considering that the "unsaved changes" message (and discard button) refers to the same provisional draft, regardless of context (i.e. slideout vs. edit page), I think it'd make a lot of sense to have them render in roughly the same position within the respective UIs.

![CleanShot 2022-02-18 at 17 27 19](https://user-images.githubusercontent.com/298510/154724517-1021e5b1-8d60-47ff-8cfd-6ef3fc645f77.png)

**Comment by thupsi** [comment]
Hmm, maybe the whole UI for saving etc could be moved to the top? It would be weird to separate the notification/discard UI from the rest.

**Reply by mmikkel** [reply]
Maybe! ...the fact that the "Cancel" and "Save" buttons is at the bottom is perhaps why the "Showing your unsaved changes" UI was placed at the bottom as well – i.e. to keep all the "actions" in the same general area, like they are on the edit page. 

Somehow though, in terms of consistency and intuitiveness I feel like keeping all of these elements in the same general area is less of a concern than the ones I outlined above. To me, the slideout's "Cancel" and "Save" actions feel like they belong in their current position, probably because that's a pretty typical UI pattern for modals – which technically the slideouts still are. And while true that edit pages have both the "Unsaved changes" message and the entry actions at the top of the pages, visually they're clearly separate. So, I'm pretty sure moving the "Unsaved changes" part to the top bar, whilst keeping the Cancel/Save buttons at the bottom, would hit the sweet spot in terms of consistency and intuitiveness.

**Comment by brandonkelly** [comment]
The space you’re pointing to is where the tabs go, so if we move the message up there, we’re talking about doubling the height of the header, which would suck on mobile, unless we also move the **Save** and **Cancel** buttons up to the top as well.

I think it’s good to have the message / **Discard** button close to the other action buttons, so moving them all is probably my preference. But it would be a break in consistency with all other modals/slideouts/HUDs where the action buttons are always at the bottom.

**Reply by mmikkel** [reply]
Good point, I didn’t consider tabs!

**Reply by brandonkelly** [reply]
If the main concern is just that the message will go unnoticed, we could explore making it stand out a bit more… something like:

<img width="859" alt="An element editor slideout footer, with a subtle blue gradient behind the “Showing your unsaved changes” message" src="https://user-images.githubusercontent.com/47792/154742915-054fc6f0-b845-4ae4-a494-a266bfc03471.png">

**Reply by mmikkel** [reply]
It going unnoticed was definitely my main concern - so giving it a bit more visual weight could absolutely go a long way.

**Reply by brandonkelly** [reply]
Done (33a3dec2cacbaa925ddfb22d399e610cce153b2b). Pretty happy with it:

<img width="834" alt="Revised “Showing your unsaved changes” notice" src="https://user-images.githubusercontent.com/47792/154754003-e066fa2a-4582-4b70-85cd-ab1ab86a7869.png">

**Reply by mmikkel** [reply]
That should help! Thanks.
